torch/quantization/observer.py

```diff
@@ -799,6 +802,129 @@ def forward(self, x_orig):
+    max_inputs_col: torch.Tensor
+    min_weights_col: torch.Tensor
+    max_weights_col: torch.Tensor
+    min_weights_row: torch.Tensor
+    max_weights_row: torch.Tensor
+
+    def __init__(self, dtype=torch.quint8,
+                 qscheme=torch.per_channel_affine, reduce_range=False,
+                 quant_min=None, quant_max=None, factory_kwargs=None) -> None:
+        super(InputWeightMinMaxObserver, self).__init__(ch_axis=1, dtype=dtype,
+                                                        qscheme=qscheme,
+                                                        reduce_range=reduce_range,
+                                                        quant_min=quant_min,
+                                                        quant_max=quant_max,
+                                                        factory_kwargs=factory_kwargs)
+
+        # We want to get the columns of the weights and inputs
+        factory_kwargs = torch.nn.factory_kwargs(factory_kwargs)
+        self.register_buffer('min_inputs_col', torch.tensor([], **factory_kwargs))
+        self.register_buffer('max_inputs_col', torch.tensor([], **factory_kwargs))
+        self.register_buffer('min_weights_col', torch.tensor([], **factory_kwargs))
+        self.register_buffer('max_weights_col', torch.tensor([], **factory_kwargs))
+        self.register_buffer('min_weights_row', torch.tensor([], **factory_kwargs))
+        self.register_buffer('max_weights_row', torch.tensor([], **factory_kwargs))
+
+    def forward(self, x_orig, w_orig):
+        if not (x_orig.ndim == 2 and w_orig.ndim == 2 and x_orig.shape[1] == w_orig.shape[1]):
+            raise ValueError(
+                "Input and Weight must have the same column dimension. " +
+                f"Found {x_orig.shape} and {w_orig.shape} instead."
+            )
+
+        return self._forward(x_orig, w_orig)
+
+    def _forward(self, x_orig, w_orig):
+        r"""
+        Calculates the min/max values of each input column, weight column, and weight row.
+        """
+        if w_orig.numel() == 0:
+            return (x_orig, w_orig)
+
+        # Calculate the min/max of input columns
+        self.ch_axis = 1
+        min_inputs_col, max_inputs_col = self._find_minmax(x_orig, self.min_inputs_col, self.max_inputs_col)
+        self.min_inputs_col.resize_(min_inputs_col.shape)
+        self.max_inputs_col.resize_(max_inputs_col.shape)
+        self.min_inputs_col.copy_(min_inputs_col)
+        self.max_inputs_col.copy_(max_inputs_col)
+
+        # Calculate the min/max of weight columns
+        min_weights_col, max_weights_col = self._find_minmax(w_orig, self.min_weights_col, self.max_weights_col)
+        self.min_weights_col.resize_(min_weights_col.shape)
+        self.max_weights_col.resize_(max_weights_col.shape)
+        self.min_weights_col.copy_(min_weights_col)
+        self.max_weights_col.copy_(max_weights_col)
+
+        # Calculate the min/max of weight rows
+        self.ch_axis = 0
```

Comment:
rather than doing this, it might be a better idea to take an axis argument in _find_minmax